src/call_frame.rs

```diff
@@ -32,16 +37,26 @@ pub struct Context {
 }
 
 impl Context {
-    pub fn new(program_code: Vec<U256>, gas_stipend: u32) -> Self {
+    pub fn new(
+        program_code: Vec<U256>,
+        gas_stipend: u32,
+        address: Address,
```

Comment:
Whose address is this? The contract's?

Reply:
Correct.

Reply:
I'd suggest adding that information to the name: `contract_address`.